### DISCUSSION TITLE
[minor] WTSHashMap::remove() 有可能的 NULL 问题。

### DISCUSSION THREAD

**Discussion by hyperdreamer** [wondertrader/wondertrader#206] [想法/Ideas]
inline void add(const T& _key, WTSObject* obj, bool bAutoRetain = true)
    {
        if (bAutoRetain && obj) obj->retain();
     
        WTSObject* pOldObj = NULL;
        auto it = _map.find(_key);
        if (it != _map.end()) pOldObj = it->second;
        if (pOldObj) pOldObj->release();
     
        _map[_key] = obj;   // obj can be NULL
    }

因为 NULL 有可能被加入，因此在删除时是不是也应该加入检查会比较安全。就像 WTSMap::remove() 一样？改为：

    inline void remove(const T& _key)
    {
        auto it = _map.find(_key);
        if (it != _map.end()) {
            WTSObject* obj = it->second;
            _map.erase(it);
            if (obj) obj->release();
        }
    }

P.S.: WTSMap::clear(),  WTSHashMap::clear(), WTSQueue::front(), WTSQueue::back(), WTSQueue::clear() 是不是也有这个问题？

**Comment by wondertrader** [comment]
![image](https://github.com/wondertrader/wondertrader/assets/62869597/08a3927c-55a7-42df-a8de-5f83dad5280f)
这一句就是对null的检查啊

**Reply by hyperdreamer** [reply]
抱歉，这句是我自己改的。原代码是：

	inline void remove(const T &_key)
	{
		auto it = _map.find(_key);
		if(it != _map.end())
		{
			it->second->release();
			_map.erase(it);
		}
	}

**Comment by wondertrader** [comment]
我明白你的意思了

**Comment by wondertrader** [comment]
最新dev改进去，谢谢了
